src/pybind/mgr/telegraf/basesocket.py

```diff
@@ -0,0 +1,38 @@
+import socket
+
+
+class BaseSocket(object):
+    schemes = {
+        'unixgram': (socket.AF_UNIX, socket.SOCK_DGRAM),
+        'unix': (socket.AF_INET, socket.SOCK_STREAM),
+        'tcp': (socket.AF_INET, socket.SOCK_STREAM),
+        'tcp6': (socket.AF_INET6, socket.SOCK_STREAM),
+        'udp': (socket.AF_INET, socket.SOCK_DGRAM),
+        'udp6': (socket.AF_INET6, socket.SOCK_DGRAM),
+    }
+
+    def __init__(self, url):
+        self.url = url
+
+        try:
+            socket_family, socket_type = self.schemes[self.url.scheme]
+        except KeyError:
+            raise RuntimeError('Unsupported socket type: %s', self.url.scheme)
+
+        self.sock = socket.socket(family=socket_family, type=socket_type)
+        if self.sock.family == socket.AF_UNIX:
+            self.address = self.url.path
+        else:
+            self.address = (self.url.hostname, self.url.port)
+
+    def connect(self):
+        return self.sock.connect(self.address)
+
+    def close(self):
+        self.sock.close()
+
+    def send(self, data, flags=0):
+        return self.sock.send(data.encode('utf-8') + b'\n', flags)
+
+    def __del__(self):
```

Comment:
nit, could implement the `__enter__()` and `__exit__()` magic methods, and always the `with` block.

Reply:
Done!